apps/user_status/lib/Db/UserStatusMapper.php

```diff
@@ -125,12 +126,31 @@ public function findByUserIds(array $userIds):array {
 		return $this->findEntities($qb);
 	}
 
+	/**
+	 * @param int $olderThan
+	 * @param int $now
+	 */
+	public function clearStatusesOlderThan(int $olderThan, int $now): void {
+		$qb = $this->db->getQueryBuilder();
+		$qb->update($this->tableName)
+			->set('status', $qb->createNamedParameter(IUserStatus::OFFLINE))
+			->set('is_user_defined', $qb->createNamedParameter(false, IQueryBuilder::PARAM_BOOL))
+			->set('status_timestamp', $qb->createNamedParameter($now, IQueryBuilder::PARAM_INT))
+			->where($qb->expr()->lte('status_timestamp', $qb->createNamedParameter($olderThan, IQueryBuilder::PARAM_INT)))
+			->andWhere($qb->expr()->orX(
+				$qb->expr()->eq('is_user_defined', $qb->createNamedParameter(false, IQueryBuilder::PARAM_BOOL)),
```

Comment:
```suggestion
				$qb->expr()->eq('is_user_defined', $qb->createNamedParameter(false, IQueryBuilder::PARAM_BOOL), IQueryBuilder::PARAM_BOOL),
```

`eq` takes an optional type as well

Reply:
Done